core/src/KokkosExp_MDRangePolicy.hpp

```diff
@@ -314,7 +315,44 @@ struct MDRangePolicy : public Kokkos::Impl::PolicyTraits<Properties...> {
   }
   bool impl_tune_tile_size() const { return m_tune_tile_size; }
 
+  tile_type tile_size_recommended() const {
```

Comment:
Can this be `static`?

Reply:
It looks like the recommended tile size for the last rank depends on the input dimension, which partially contradicts what I described in the [previous comment](https://github.com/kokkos/kokkos/pull/6839/files/6d6bc339b2cc973b49c69980fa2eb5f01a6139b0#r1688821503).
```
    if (tile_rank == last_rank) {
      rec_tile_size = tile_size_last_rank(
          properties, m_upper[last_rank] - m_lower[last_rank]);
    }
```
`max_total_tile_size` could be made into a `static` function, but for consistency, I think it makes sense to either have both `tile_size_recommended` and `max_total_tile_size` be static or have neither be.

Reply:
Fair enough.